bitcoin/src/blockdata/weight.rs

```diff
@@ -33,14 +33,24 @@ impl Weight {
     pub const fn from_wu(wu: u64) -> Self { Weight(wu) }
 
     /// Constructs `Weight` from kilo weight units returning `None` if overflow occurred.
-    pub fn from_kwu(wu: u64) -> Option<Self> { wu.checked_mul(1000).map(Weight) }
+    pub const fn from_kwu(kwu: u64) -> Option<Self> {
+        match kwu.checked_mul(1000) {
+            Some(weight) => Some(Weight(weight)),
+            None => None,
+        }
```

Comment:
Kind of sad, this feels like a coding regression to not be able to use `map`. I guess that is a language problem.

Reply:
Indeed. :( Will get fixed in maybe 10 years...

Reply:
May be worth commenting something to the effect of "can use `map` when MSRV allows it in constfns". Maybe not. Maybe we should just audit all the constfns every time we do a constfn.

Reply:
Such a comment seems noisy but on the other hand if it includes "MSRV" in it then it may remind us at MSRV bump time to do it since grepping for MSRV is likely to happen at that time.

Reply:
These code snippets also aren't horrible and usually contained in a very small method. So I wouldn't sweat it too much.

Reply:
That MSRV bump will come in maybe 5 years. `const FnOnce` trait needs to exists before that happens and there's currently hot debate going on about how to do it. It won't stabilize anytime soon. So "MSRV" is misleading, it's a language limitation.